tiledb/CMakeLists.txt

```diff
@@ -545,10 +545,15 @@ if(APPLE)
   endif()
 endif()
 
-# On Linux, must explicitly link -lpthread -ldl in order for static linking
-# to libzstd.
+# Enable threading library for systems where it is not automatically enable.
+find_package(Threads REQUIRED)
+if (CMAKE_THREAD_LIBS_INIT)
```

Comment:
Is `CMAKE_THREAD_LIBS_INIT` ever false if `find_package` returns without error for a required package? If so, in what circumstances?

Also, if we still need this variable, we should document that it's set by `find_package`; one of the many non-obvious dependencies of CMake.

Reply:
CMAKE_THREAD_LIBS_INIT is false if threading is supported, but no compiler flag or library is needed to set it (e.g. WIN32 builds). I'll add a comment here and at the other location it is used.